src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -187,12 +185,15 @@ AssemblyMetadata GetMetadataWorker()
         return;
 
         void GetStorageInfoFromTemporaryStorage(
-            FileKey moduleFileKey, out TemporaryStorageService.TemporaryStreamStorage storage, out UnmanagedMemoryStream stream)
+            FileKey moduleFileKey, out TemporaryStorageIdentifier storageIdentifier, out UnmanagedMemoryStream stream)
         {
             int size;
+
+            // Create a temp stream in memory to copy the metadata bytes into.
             using (var copyStream = SerializableBytes.CreateWritableStream())
```

Comment:
i tried to beef up the comments to make the flow of data clearer.  It's a little obtuse by it always tends to follow the same pattern in a few places:

1. start with original value.
2. serialize into in-memory-stream.  
3. ask temp-storage-system to dump that stream to a memory-mapped file.
4. get a handle/identifier back to that dumped info
5. read the dumped data back into some value (now backed by unmoving, unreleased, memory-mapped-data.